torch/nested/_internal/ops.py

```diff
@@ -380,8 +382,31 @@ def prod_dim_int(func, *args, **kwargs):
     return NestedTensor(func(inp._values, **new_kwargs), **extract_kwargs(args[0]))
 
 
+# Ordinarily, if a program calls .tolist compiling still works because there is
+# special handling in dynamo, but for tensor subclasses if .tolist is called
+# inside torch dispatch, the .tolist call will be directly on a FakeTensor.
+# This would result in an error since wrapper subclasses don't have storage.
+# To avoid this, we handle the fake tensor case by (1) specializing on the size
+# of the tensor to create the output Python list, and (2) creating unbacked
+# symints for each element of the list.
```

Comment:
Why can't we just put this on a FakeTensor `tolist` implementation?

Reply:
I'm not sure FakeTensor can intercept this since its not an aten op

Reply:
Literally put it as a method on FakeTensor lol

Reply:
ohh good point!